components/builder-depot-client/src/lib.rs

```diff
@@ -319,9 +321,7 @@ impl Client {
     ///
     /// * Key cannot be found
     /// * Remote Builder is not available
-    pub fn schedule_job<I>(&self, ident: &I, package_only: bool, token: &str) -> Result<(String)>
-    where
-        I: Identifiable,
+    pub fn schedule_job(&self, ident: &PackageIdent, package_only: bool, token: &str) -> Result<(String)>
```

Comment:
@christophermaier Can you shed some more light on why we want to remove the use of `Identifiable`?  I think I missed that discussion.

Reply:
@raskchanky I've been taking another long look at our primary `PackageIdent` Rust type (in the core library) and have realized that we have lots of places in our codebase where we take something that could *act* like an ident and implement this trait on it, giving us some minimal common behavior like `.origin()`, `.release()`, `.fully_qualified()`, etc.

I suspect that if we were starting from scratch today, it would be easier to simply convert these ident-like structs into a legit. `PackageIdent` so we get all of the desired behavior (i.e. like one `Display` representation, one implementation of `.archive_name()`, .etc.). While this isn't proven out yet, I'm hoping that we're going to find more places where we've built ident-like structs to solve a local problem (like a Protobuf representation, a value struct for serializing or deserializing to TOML/JSON/etc) but then leaked those ident-like structs back out to consuming code that might simply be expecting a `PackageIdent` at the end of the day. When I was looking at what `Identifiable` was providing for us, and after subtracting the behavior that can be tacked directly back on `PackageIdent`, then `Identifiable` has much less purpose in such a world.

Anyway, it might be worth having a chat about this with a few other folks to feel it out. It's been feeling like a positive direction, but not if it ends up being at odds with assumptions made or directions taken elsewhere.